packages/components/src/Form/Form.stories.mdx

```diff
@@ -0,0 +1,154 @@
+import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
+import { useRef, useState } from "react";
+import { useFormState } from "@jobber/hooks";
+import { Form } from "@jobber/components/Form";
+import { Content } from "@jobber/components/Content";
+import { Card } from "@jobber/components/Card";
+import { InputText } from "@jobber/components/InputText";
+import { Button } from "@jobber/components/Button";
+
+<Meta title="Components/Form" component={Form} />
+
+# Form
+
+The Form component is a wrapper component that handles the submission and
+validation of forms.
+
+For more information about `validations` using any of the `Input` components,
+see the [InputText](/components/input-text/#validation-message) documentation.
+
+```ts
+import { Form } from "@jobber/components/Form";
+```
+
+<Canvas>
+  <Story name="Form">
+    {args => {
+      const [{ isDirty, isValid }, setFormState] = useFormState();
+      const [first, setFirst] = useState("");
+      return (
+        <Form
+          {...args}
+          onSubmit={() => alert("Submitted 🎉🎉🎉")}
+          onStateChange={setFormState}
+        >
+          <Content>
+            <InputText
+              placeholder="First Name"
+              name="firstName"
+              value={first}
+              onChange={setFirst}
+              validations={{
+                required: {
+                  value: true,
+                  message: "Tell us your name",
+                },
+                minLength: {
+                  value: 3,
+                  message: "Your name is too short.",
+                },
+              }}
+            />
+            <InputText
+              placeholder="Last Name"
+              name="lastName"
+              validations={{
+                required: {
+                  value: true,
+                  message: "Tell us your last name.",
+                },
+              }}
+            />
+            <Button
+              label="Submit Form"
+              submit={true}
+              disabled={!isDirty || !isValid}
+            />
+          </Content>
+        </Form>
+      );
+    }}
+  </Story>
+</Canvas>
+
+<ArgsTable of={Form} story="Form" />
+
+## onStateChange
+
+`onStateChange` will call whenever the state of the form is changed. It is best
+practice to use the [`useFormState`](/hooks/useFormState) hook with the
+`onStateChange` prop.
+
+<Canvas withSource="none">
+  <OnStateChangeExample />
```

Comment:
I think this can be updated to use `ExampleWithHooks` now. We could also do it in a follow up PR